### PR TITLE
computing the section radius and use it to cull the them when rendering.

### PR DESCRIPTION
**Pull request by jantlo**
computing the section sphere's square radius and using it reject those sections behind the camera.

### CODE REVIEW

File: src/wipeout/track.c.orig [hunk 1]
```diff
@@ -0,0 +1,392 @@
+#include "../mem.h"
```
Comment (The-King-of-Toasters) [inline src/wipeout/track.c.orig:1]: diff fail?

**Comment by phoboslab** [comment]
This just fixes relying on the `2048` constant for the section size, right? Please get rid of the `track.c.orig`; otherwise LGTM.

**Comment by jantlo** [comment]
Yes, it computes the square radius per section and uses it instead the 2048 value. 
It requires some additional computation, like the scale_factor (based on the current FOV) applied over the section's radius to compute properly whether the section is visible or not.

File: src/wipeout/track.c.orig (hunk 1 above)
Comment (jantlo) [reply src/wipeout/track.c.orig]: i guess some git rebase failed and by mistake I committed this .orig file. Next commit removes it.

**Comment by jantlo** [comment]
BTW: is it allow by github to squash the changes locally and re-push the branch again, (using --force) ?

**Comment by phoboslab** [comment]
> requires [...] scale_factor (based on the current FOV) applied over the section's radius to compute properly whether the section is visible or not.

I don't understand. Why? This isn't doing any frustum culling, based on the actual FOV, right? It just checks if the section is behind or in front of the camera, like before. Why's the scale_factor needed here?

Also, it doesn't seem to work with extrem FOVs:
Screenshot_2023-09-09_15-26-59

**Comment by jantlo** [comment]
Uhm, thanks for the finding, computing the horizontal fov was wrong.
yes, it is only removing all the sections behind the camera, but the radius is not enough, the radius has to be scaled based on the max FOV (for safety) otherwise some sections are culled to early. Check the drawing (so you can see my skills with MS Paint):
image

As you can see there are case where the section's sphere is behind the camera and distance to the camera apex is the radius and it is still partially inside. So this scale factor computes the minimum distance required to ensure is fully outside.

**Comment by tomaz82** [comment]
This feels very, what's the word, unconventional. Why not use proper frustum culling using 5 frustum planes and bounding spheres for the sections? That would lower the amount of sections being rendered even further and the math is also probably faster than this.

**Comment by jantlo** [comment]
Well, it is, it was just the next step to just remove the sections behind the camera using the radius.  And it is straightforward but it requires to handle these cases, and it is a matter of computing the this scale factor for the worse case. 

Maybe a simpler approach is to just discard those sections which distance to the camera forward plane is bigger than the its radius, but it is just less accurate. Or just go for the usual frustum culling.

**Comment by phoboslab** [comment]
Thanks for the explanation of why this is necessary!

I'm siding with @tomaz82 here, though. The current solution works just as well (maybe drawing 1-2 sections more than necessary). This PR introduces some more complexity with no real benefit.

Aside, this this also adds some functions that need to be replicated in every (potential) render backend; maybe we should have a common `render.c` that has the common functions shared between all backends.

**Comment by jantlo** [comment]
Ok, no problem, it was more an exercise of correctness than a real improvement.

**Pull request closed without merging**